### ISSUE TITLE
Implement the RemoteSettings SignatureVerifier

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/uniffi-rs#836]
[https://github.com/mozilla-mobile/android-components/blob/master/components/service/experiments/src/main/java/mozilla/components/service/experiments/SignatureVerifier.kt](https://github.com/mozilla-mobile/android-components/blob/master/components/service/experiments/src/main/java/mozilla/components/service/experiments/SignatureVerifier.kt)

Ref: [https://github.com/Vishwa-Mozilla/remote-settings-client](https://github.com/Vishwa-Mozilla/remote-settings-client)

**Comment by data-sync-user** [comment]
➤ Edouard Oger commented:

There was some desire to implement this in Rust, is this still the case?

**Comment by data-sync-user** [comment]
➤ Vlad Filippov commented:

eoger it should be in https://github.com/mozilla-services/remote-settings-client

**Comment by data-sync-user** [comment]
➤ Edouard Oger commented:

I'm aware of that project, however the crypto parts are implemented using OpenSSL which isn't going to work for us since the code will ship alongside gecko. If we decide to use this crate, we'll need to put in some work ( https://docs.google.com/document/d/1axiv5yc4aoEIGhjYYC2SgN1-TF1B03MKSnO9j0jISuY ) on rc_crypto to support these signature verification operations.

**Comment by data-sync-user** [comment]
➤ Vlad Filippov commented:

eoger ah it might not have been completed then. @Vishwa-Mozilla has been working on adding this but maybe ran out of time

**Comment by data-sync-user** [comment]
➤ Edouard Oger commented:

So I assume we still want this in Rust?

**Comment by data-sync-user** [comment]
➤ Vlad Filippov commented:

> So I assume we still want this in Rust?

Yeap!

**Comment by data-sync-user** [comment]
➤ Ryan Kelly commented:

Regrouping on some context here:

  * I believe the required crypto primitives for this were landed in rc_crypto in https://github.com/mozilla/application-services/pull/3557.
  * Next step would be to add a verifier backend in `remote-settings-client` ( https://github.com/mozilla-services/remote-settings-client ) that uses rc_crypto rather than OpenSSL.
  * (This might have the side-effect of causing us to discover other missing pieces in rc_crypto).

  * When that's ready, we could switch nimbus-sdk to use remote-settings-client rather than talking directly to the remote-setting server via HTTP.

**Comment by data-sync-user** [comment]
➤ Mathieu Leplatre commented:

I started to work on this (https://github.com/mozilla-services/remote-settings-client/pull/50)

I started by replacing openssl by x509-parser

  * ring.  But I'm stuck on making it work. I suspect I have to decode/encode the signature content but I can't really figure it out. I have the equivalent working in Python and summarized it here https://stackoverflow.com/questions/66492119/unspecified-error-when-verifying-ecdsa-signature-with-ring  Update! Got it working :tada:   https://github.com/mozilla-services/remote-settings-client/pull/54

Then I tried with rc_crypto because, from I can see in the ECDSA tests, nothing seems to be necessary on splitting/decoding the signature bytes. But then was stuck on compiling it on my system (:roll_eyes: ). I opened https://github.com/mozilla/application-services/issues/3908 for that 😀
